File: packages/wasm/as/assembly/__tests__/invoke.spec.ts
import { w3_add_invoke, w3_invoke } from "../";

function methodName(_input: ArrayBuffer): ArrayBuffer {
  return new ArrayBuffer(0);
}

describe("Invoke: Sanity", () => {
  it("Compiles", () => {
    w3_add_invoke("methodName", methodName);
    w3_invoke(1, 1);
  });
});

File: packages/wasm/rs/tests/invoke_spec.rs
use polywrap_wasm_rs::invoke::{w3_invoke, InvokeArgs};

pub fn method_name(input: &[u8]) -> Vec<u8> {
    input.to_vec()
}

#[test]
pub fn it_compiles_sanity() {
    let invoke_args = InvokeArgs {
        method: "method_name".to_string(),
        args: vec![0x1, 0x2, 0x3],
    };
    assert!(w3_invoke(invoke_args, Some(method_name)))
}
